common/src/main/java/tachyon/LeaderSelectorClient.java

```diff
@@ -62,11 +62,18 @@ public LeaderSelectorClient(String zookeeperAddress, String electionPath, String
 
     // Create a leader selector using the given path for management.
     // All participants in a given leader selection must use the same path.
-    // ExampleClient here is also a LeaderSelectorListener but this isn't required.
     CuratorFramework client =
         CuratorFrameworkFactory.newClient(mZookeeperAddress, new ExponentialBackoffRetry(
             Constants.SECOND_MS, 3));
     client.start();
+
+    // Sometimes, if the master crashes and restarts too quickly, zookeeper thinks the new client is
+    // still an old one. Close the "old" client and recreate a new one.
+    client.close();
```

Comment:
zk has a timeout mechanism, but it is a few seconds. Therefore, if the master restarts in less than a few seconds, it gets into a strange state.

I extracted client creation into a method.